packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1939,6 +1949,15 @@ class RenderEditable extends RenderBox with RelayoutWhenSystemFontsChangeMixin,
       return moveSelectionRightByLine(cause);
     }
 
+    // If greatest edge is already at the end of a line, don't do anything.
+    final int greatestOffset = math.max(selection!.baseOffset, selection!.extentOffset);
+    final TextSelection currentLine = _getLineAtOffset(TextPosition(
```

Comment:
Could the problem be we're initializing a new `TextPosition` with the default text affinity being downstream? If text affinity is upstream then `_getLineAtOffset` won't go to the next line right?